src/commands/basecommand.cc

```diff
@@ -175,6 +175,13 @@ int ZypperBaseCommand::defaultSystemSetup( Zypper &zypper, SetupSystemFlags flag
     init_repos( zypper );
     if ( zypper.exitCode() != ZYPPER_EXIT_OK )
       return zypper.exitCode();
+    else if ( flags_r.testFlag( FailIfReposFail ) && zypper.exitInfoCode() != ZYPPER_EXIT_OK ) {
+      zypper.out().taggedPar( 4, MSG_ERRORString( command()[0]+":" ), _("Due to the treatment of orphaned packages dist-upgrade depends on a proper repository setup more than any other command. It must not continue if enabled repositories fail to refresh. This may severely damage the system. If a failing repository is actually not needed, it must be disabled.")
+      +std::string(" ")
+      +_("See 'man zypper' for more information about this command.") );
```

Comment:
Should this be really inside basecommand? The wording of the error message suggests that is a error which can happen only in dup.  This could be completely handled in DistUpgradeCmd::systemSetup. Since its overloaded already a simple if() checking the return code of defaultSystemSetup should be enough?

Reply:
Also shouldn't this check for ZYPPER_EXIT_INF_REPOS_SKIPPED rather then  != ZYPPER_EXIT_OK ?